### ISSUE TITLE
[java] LinguisticNaming should support AtomicBooleans

### ISSUE THREAD

**Issue by SamCarlberg** [pmd/pmd#1334]
**Affects PMD Version:** 
6.7.0

**Rule:**
LinguisticNaming

**Description:**
Fields of type `AtomicBoolean` are prohibited from following the naming patterns `isX`, `shouldX`, `hasX`, etc., e.g. `isFlagSet`. In my view, AtomicBooleans are simply threadsafe wrappers around `boolean` values and are therefore perfectly valid to have a boolean-style name prefix.

**Code Sample demonstrating the issue:**

All of the following fields will trigger the rule:
```java
private final AtomicBoolean isX = new AtomicBoolean(true);
private final AtomicBoolean hasX = new AtomicBoolean(true);
private final AtomicBoolean canX = new AtomicBoolean(true);
private final AtomicBoolean haveX = new AtomicBoolean(true);
private final AtomicBoolean willX = new AtomicBoolean(true);
private final AtomicBoolean shouldX = new AtomicBoolean(true);
```

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
@SamCarlberg sounds very reasonable, thanks for the feedback

**Comment by adangel** [comment]
:+1: Yes, that absolutely makes sense. We should treat AtomicBoolean the same way as boolean for this rule.
